Mathlib/RingTheory/LocalProperties/Flat.lean

```diff
@@ -0,0 +1,214 @@
+    obtain ⟨⟨m, sm⟩, eqy⟩ := mk'_surjective J.primeCompl (mkLinearMap _ M) y
+    simp only [Function.uncurry_apply_pair, ← mk_eq_mk'] at eqx eqy
+    simp only [← eqx, ← eqy, map'_mk, eqv'_mk_apply, LinearMap.rTensor_tmul]
+  have inj : Function.Injective ((eqv N' M J.primeCompl).toLinearMap ∘ₗ g1) := by
+    simp only [coe_comp, LinearEquiv.coe_coe, EmbeddingLike.comp_injective]
+    exact ((Module.Flat.iff_rTensor_preserves_injective_linearMap' _ _).mp (h J hJ)
+      (map J.primeCompl f) inj)
+  rw [this] at inj
+  simp only [coe_comp, LinearEquiv.coe_coe, EquivLike.injective_comp] at inj
+  exact inj
+
+variable (s : Finset R) (spn : span (s : Set R) = ⊤)
+include spn
+
+theorem Flat_of_localization_finitespan  (h : ∀ r : s, Module.Flat
+    (Localization (Submonoid.powers r.1)) (LocalizedModule (Submonoid.powers r.1) M)) :
+    Module.Flat R M := by
+  apply (Module.Flat.iff_rTensor_preserves_injective_linearMap R M).mpr
+  intro N N' _ _ _ _ f finj
+  apply injective_of_localization_finitespan s spn
+  intro r
+  have inj : Function.Injective (map (Submonoid.powers r.1) f) := by
+    unfold LocalizedModule.map mapExtendScalars
+    simp only [coe_comp, LinearEquiv.coe_coe, Function.comp_apply, LinearMap.coe_mk,
+      LinearEquiv.restrictScalars_apply, extendScalarsOfIsLocalizationEquiv_apply,
+      restrictScalars_extendScalarsOfIsLocalization, AddHom.coe_mk, ← ker_eq_bot, coe_toAddHom]
+    rw [ ← localized'_ker_eq_ker_localizedMap, ker_eq_bot.mpr finj, localized'_bot]
+  set g1 := (rTensor (LocalizedModule (Submonoid.powers r.1) M)
+    ((LocalizedModule.map (Submonoid.powers r.1)) f))
+  set g2 := ((LocalizedModule.map (Submonoid.powers r.1)) (rTensor M f))
+  have : (eqv N' M (Submonoid.powers r.1)) ∘ₗ g1 = g2 ∘ₗ (eqv N M (Submonoid.powers r.1)) := by
+    apply TensorProduct.ext'
+    intro x y
+    unfold g1 g2 eqv lmap
+    simp only [ofLinear_toLinearMap, coe_comp, Function.comp_apply, LinearMap.rTensor_tmul,
+      extendScalarsOfIsLocalization_apply', LinearEquiv.coe_coe]
+    obtain ⟨⟨n, sn⟩, eqx⟩ := mk'_surjective (Submonoid.powers r.1) (mkLinearMap _ N) x
+    obtain ⟨⟨m, sm⟩, eqy⟩ := mk'_surjective (Submonoid.powers r.1) (mkLinearMap _ M) y
+    simp only [Function.uncurry_apply_pair, ← mk_eq_mk'] at eqx eqy
+    simp only [← eqx, ← eqy, map'_mk, eqv'_mk_apply, LinearMap.rTensor_tmul]
+  have inj : Function.Injective ((eqv N' M (Submonoid.powers r.1)).toLinearMap ∘ₗ g1) := by
+    simp only [coe_comp, LinearEquiv.coe_coe, EmbeddingLike.comp_injective]
+    exact ((Module.Flat.iff_rTensor_preserves_injective_linearMap' _ _).mp (h r)
+      (map (Submonoid.powers r.1) f) inj)
+  rw [this] at inj
+  simp only [coe_comp, LinearEquiv.coe_coe, EquivLike.injective_comp] at inj
+  exact inj
+end flatlocal
+
+section flatifflocal
+
+variable (R R' : Type*) [CommRing R] [CommRing R'] [Algebra R R'] (S : Submonoid R)
+  [IsLocalization S R']
+include S
+instance : Module.Flat R (Localization S) := by
```

Comment:
Sorry but this is already added into Mathlib, by myself: [Localization.flat](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Flat/Localization.html#Localization.flat). See [discussions on Zulip](https://leanprover.zulipchat.com/#narrow/channel/116395-maths/topic/Puzzles.20around.20Picard.20group/near/480318967).